dali/operators/random/rng_base_gpu.h

```diff
@@ -106,64 +105,57 @@ std::pair<std::vector<int>, int> DistributeBlocksPerSample(
   return {sizes, max_blocks};
 }
 
-template <typename T>
-int64_t SetupBlockDescs(BlockDesc<false> *blocks, int64_t block_sz, int64_t max_nblocks,
-                        TensorListView<StorageGPU, T> &output,
-                        TensorListView<StorageGPU, const T> &input) {
-  (void) input;
+template <int ndim>
+int64_t SetupBlockDescs(BlockDesc *blocks, int64_t block_sz, int64_t max_nblocks,
+                        const TensorListShape<ndim> &shape, int channel_dim = -1) {
   std::vector<int> blocks_per_sample;
   int64_t blocks_num;
-  auto &shape = output.shape;
   std::tie(blocks_per_sample, blocks_num) = DistributeBlocksPerSample(shape, block_sz, max_nblocks);
   int64_t block = 0;
   for (int s = 0; s < shape.size(); s++) {
-    T *sample_data = static_cast<T *>(output[s].data);
     auto sample_size = volume(shape[s]);
```

Comment:
You can extract the TensorShape here and doctor the channel dimension here instead of relying on the caller passing a manipulated input shape.
```suggestion
    shape_in_pixels = shape[s];
    if (channel_dim >= 0)
      shape_in_pixels[channel_dim] = 1;l
    auto sample_size = volume(shape_in_pixels);
```

Reply:
Will do